### PR TITLE
Add Array.prototype.flat polyfill

### PR DESCRIPTION
**Pull request by jonathantneal**
This adds the `flat` method for Arrays.

Resolves #4294

### CODE REVIEW

**Comment by jonathantneal** [comment]
I think I agree to everything, but I could use 2 small clarifications.

> All contributions to the polyfill service are made under the MIT licence

1. By agreeing to this, am I retroactively relicensing all of my other contributions to this repository? I think the answer is “no”.
2. Do I need to license my work to myself first, and then submit it to this repository? I think the answer is “that’s up to me”, and so if I wish to CC0 license my work, I need to first do that in `config.json`, similar to the `URL` polyfill.

I apologize for being sheepish with licenses. It’s one of the reasons I’ve always used CC0.

File: packages/polyfill-library/polyfills/Array/prototype/flat/polyfill.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+			--depth;
+
+			Array.prototype.push.apply(stack, next);
+		} else {
+			result.unshift(next);
+		}
+	}
+
+	return result.concat(stack);
+});
```
[the remark this answers is not in the document]
Comment (jonathantneal) [reply packages/polyfill-library/polyfills/Array/prototype/flat/polyfill.js:19]: @caub, thanks for looking at my code. How would you handle the secondary depth argument? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat#Syntax

Perhaps that is why my solution is more involved? Also, I will gladly update this PR with any and all superior solutions. 😄

[the remark this answers is not in the document]
Comment (caub) [reply packages/polyfill-library/polyfills/Array/prototype/flat/polyfill.js:19]: oh fair enough
```js
function flat(a, depth = 1) {
  return Array.isArray(a) && depth > 0 ? a.reduce(function(b,c){ return b.concat(flat(c, depth-1)); }, []) : a;
}
```
but your approach works for the empty slots cases, so better

**Comment by JakeChampion** [comment]
Thanks for the pull-request @jonathantneal 
This feature has been implemented in the polyfill-library repo -- https://github.com/Financial-Times/polyfill-library/pull/93

**Pull request closed without merging**